fix(typed_pipes): drain thread exits silently on Broken Pipe — host detects stale session late

## What

When the app-side socket closes unexpectedly (e.g. the audio-recorder app switches devices mid-capture), the drain thread hits \`Broken pipe (os error 32)\` on \`write_frame\`, logs a warning, and exits silently. The host has no signal that the drain thread died — it only discovers the stale session the next time \`start_audio_capture\` is called with the same \`pipe_id\`. The gap is 3–21 seconds in practice.

Observed in the log:
\`\`\`
[WARN]  typed_pipes: drain write error: Broken pipe (os error 32)
[ERROR] app::audio-recorder pipe read_frame error: [Errno 9] Bad file descriptor
[INFO]  audio-recorder: capture stopped, 582 chunks
# ... 3–21 second gap ...
[WARN]  ProcessApp[audio-recorder]: AudioCapture pipe_id=audio-recorder-mic stale session — cleaning up and restarting
\`\`\`

## Reproduce

1. Open audio-recorder app
2. Start a capture, then switch audio devices while recording
3. Observe the log gap between the \`Broken pipe\` warn and the \`stale session\` warn

## Expected

The host should detect the broken drain within one render frame and immediately emit \`AudioCaptureError\` (and equivalent events for MIDI/video pipes), cleaning up the stale session proactively rather than waiting for the app to retry.

## Action Plan

- **\`src/typed_pipes.rs\`** — add \`error_flag: Arc<AtomicBool>\` to \`BinaryPipeEntry\`. Drain thread sets it before exiting with a write error. Add \`TypedPipeRegistry::drain_failed(pipe_id) -> bool\` to poll the flag.
- **\`src/process_app/routing.rs\`** — in \`poll_actions\` (or the per-frame tick path), iterate open audio/MIDI/video capture sessions and call \`drain_failed(pipe_id)\`. On true: close the pipe, remove the session from \`audio_capture_sessions\`, push \`PlexiEvent::AudioCaptureError\` so the app gets immediate notification.
- **\`src/typed_pipes.rs\`** — clear the flag in \`close()\` so it doesn't fire spuriously after a clean shutdown.

Alternative: pass a \`crossbeam_channel::Sender<(String, String)>\` (pipe_id, error) into the drain thread and let it push directly onto the host outbound event queue — more immediate, but requires routing the sender through \`open_binary\`.

## Done When

- After a Broken Pipe on the drain thread, \`AudioCaptureError\` (or equivalent) is emitted to the app within one render frame (~16ms), with no multi-second gap
- \`stale session\` warn is eliminated — the pipe is cleaned up before any retry attempt
- \`cargo test --lib\` passes

--- a/src/process_app/mod.rs
+++ b/src/process_app/mod.rs
@@ -1291,6 +1291,41 @@ impl App for ProcessApp {
             self.outbound_events.push_back(event);
         }
 
+        // Per-frame: detect audio capture pipes whose drain thread exited due
+        // to a write error (e.g. Broken pipe when the app-side socket closed
+        // unexpectedly). Emit AudioCaptureError immediately so the app is
+        // notified within one render frame rather than waiting for the next
+        // start_audio_capture retry to discover the stale session.
+        let failed_audio_pipes: Vec<String> = self
+            .audio_capture_sessions
+            .keys()
+            .filter(|pipe_id| {
+                self.pipe_registry
+                    .lock()
+                    .map(|r| r.drain_failed(pipe_id))
+                    .unwrap_or(false)
+            })
+            .cloned()
+            .collect();
+        for pipe_id in failed_audio_pipes {
+            log::warn!(
+                "ProcessApp[{}]: AudioCapture pipe_id={pipe_id} drain failed — cleaning up",
+                self.type_id
+            );
+            self.audio_capture_sessions.remove(&pipe_id);
+            self.pipe_registry
+                .lock()
+                .expect("pipe_registry poisoned")
+                .close(&pipe_id);
+            if let Ok(mut m) = self.audio_peak_meters.lock() {
+                m.remove(&pipe_id);
+            }
+            self.outbound_events.push_back(PlexiEvent::AudioCaptureError {
+                pipe_id,
+                error: "pipe drain failed (broken pipe)".to_owned(),
+            });
+        }
+
         let new_cmds = self.drain_draw_commands();
 
         for cmd in new_cmds {

--- a/src/typed_pipes.rs
+++ b/src/typed_pipes.rs
@@ -71,6 +71,11 @@ struct BinaryPipeEntry {
     /// capture callback) clone this `Arc` and push frames; the drain thread
     /// pops and writes them to the socket.
     ring: Arc<ArrayQueue<Vec<u8>>>,
+    /// Set by the drain thread before it exits due to a write error (e.g.
+    /// Broken pipe). The host polls this flag per-frame via `drain_failed()`
+    /// and emits an error event so the app is notified promptly rather than
+    /// waiting for the next retry attempt.
+    error_flag: Arc<AtomicBool>,
 }
 
 struct JsonPipeEntry {
@@ -142,9 +147,11 @@ impl TypedPipeRegistry {
 
         let ring: Arc<ArrayQueue<Vec<u8>>> = Arc::new(ArrayQueue::new(DEFAULT_RING_CAPACITY));
         let shutdown = Arc::new(AtomicBool::new(false));
+        let error_flag = Arc::new(AtomicBool::new(false));
 
         let ring_drain = Arc::clone(&ring);
         let shutdown_drain = Arc::clone(&shutdown);
+        let error_flag_drain = Arc::clone(&error_flag);
         let socket_path_drain = socket_path.clone();
         let pipe_id_log = pipe_id.clone();
 
@@ -181,6 +188,7 @@ impl TypedPipeRegistry {
                     if let Some(frame) = ring_drain.pop() {
                         if let Err(e) = write_frame(&mut writer, &frame) {
                             log::warn!("typed_pipes: drain write error: {e}");
+                            error_flag_drain.store(true, Ordering::Release);
                             break;
                         }
                     } else if shutdown_drain.load(Ordering::Acquire) {
@@ -201,6 +209,7 @@ impl TypedPipeRegistry {
             shutdown,
             drain_handle: Some(drain_handle),
             ring: Arc::clone(&ring),
+            error_flag,
         };
 
         self.pipes.insert(pipe_id.clone(), PipeEntry::Binary(entry));
@@ -261,6 +270,16 @@ impl TypedPipeRegistry {
         match self.pipes.get(pipe_id) {
             Some(PipeEntry::Binary(b)) => Some(Arc::clone(&b.ring)),
             _ => None,
+        }
+    }
+
+    /// Returns true if the drain thread for `pipe_id` exited due to a write
+    /// error (e.g. Broken pipe). The host polls this per-frame to detect pipe
+    /// failures promptly rather than waiting for the app to retry.
+    pub fn drain_failed(&self, pipe_id: &str) -> bool {
+        match self.pipes.get(pipe_id) {
+            Some(PipeEntry::Binary(b)) => b.error_flag.load(Ordering::Acquire),
+            _ => false,
         }
     }
 
@@ -354,6 +373,56 @@ mod tests {
         assert!(
             !bystander_reg.has_reader("coord-to-worker"),
             "bystander pane never opted in — has_reader must be false"
+        );
+    }
+
+    #[test]
+    fn drain_failed_set_after_broken_pipe() {
+        // Open a binary pipe, connect a client, then immediately drop it to
+        // simulate the app-side socket closing unexpectedly (Broken pipe).
+        // After the drain thread attempts a write and fails, drain_failed()
+        // must return true.
+        let mut reg = TypedPipeRegistry::new();
+        let alloc = reg
+            .open_binary("test-broken-pipe".to_string(), PipeDirection::In)
+            .expect("open_binary");
+
+        // Connect and immediately close to give the drain thread a peer that
+        // will produce Broken pipe / Connection reset on the first write.
+        {
+            let _client =
+                std::os::unix::net::UnixStream::connect(&alloc.socket_path).expect("connect");
+            // `_client` drops here, closing the far end.
+        }
+
+        // Give the drain thread time to accept the connection.
+        std::thread::sleep(std::time::Duration::from_millis(100));
+
+        // Push a frame — drain thread will attempt to write to the closed socket.
+        let ring = reg.binary_ring("test-broken-pipe").expect("ring");
+        let _ = ring.push(vec![1, 2, 3]);
+
+        // Give the drain thread time to attempt the write and set error_flag.
+        std::thread::sleep(std::time::Duration::from_millis(200));
+
+        assert!(
+            reg.drain_failed("test-broken-pipe"),
+            "drain_failed should be true after Broken pipe"
+        );
+    }
+
+    #[test]
+    fn drain_failed_false_for_healthy_pipe() {
+        let mut reg = TypedPipeRegistry::new();
+        reg.open_binary("test-healthy".to_string(), PipeDirection::In)
+            .expect("open_binary");
+        assert!(
+            !reg.drain_failed("test-healthy"),
+            "drain_failed should be false before any write error"
+        );
+        assert!(
+            !reg.drain_failed("nonexistent"),
+            "drain_failed should be false for unknown pipe"
         );
     }
 

